lms/resources/lti_launch.py

```diff
@@ -31,17 +31,25 @@ def __init__(self, request):
     def h_user(self):
         """Return the h user for the current request."""
 
+        # The h "provider" string for the current request.
+        provider = self._request.lti_user.tool_consumer_instance_guid
+
+        # The h "provider_unique_id" string for the current request.
+        provider_unique_id = self._request.lti_user.user_id
+
         def username():
             """Return the h username for the current request."""
             username_hash_object = hashlib.sha1()
-            username_hash_object.update(self.h_provider.encode())
-            username_hash_object.update(self.h_provider_unique_id.encode())
+            username_hash_object.update(provider.encode())
+            username_hash_object.update(provider_unique_id.encode())
             return username_hash_object.hexdigest()[:30]
```

Comment:
Is this moving into the user object in a different PR?

Reply:
That's coming, yes